src/api/flow.ts

```diff
@@ -210,10 +154,22 @@ export function createFlowGenerator(name: string, generator: Function) {
                 if (ret.done) return resolve(ret.value)
                 // TODO: support more type of values? See https://github.com/tj/co/blob/249bbdc72da24ae44076afd716349d2089b31c4c/index.js#L100
                 if (!ret.value || typeof ret.value.then !== "function")
-                    fail("Only promises can be yielded to asyncAction, got: " + ret)
+                    return fail("Only promises can be yielded to asyncAction, got: " + ret)
                 return ret.value.then(onFulfilled, onRejected)
             }
+
+            onFulfilled(undefined) // kick off the process
+        }) as any
+
+        res.cancel = action(`${name} - runid: ${runId} - cancel`, function() {
+            try {
+                gen.return()
```

Comment:
Check if `return` exists on the generator maybe? If a non-generator iterator is available, it's probably best to skip the call. This is a really edgy edge case though.